x-pack/legacy/plugins/reporting/export_types/csv/server/create_job.ts

```diff
@@ -4,30 +4,27 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { KibanaRequest, RequestHandlerContext } from 'src/core/server';
 import { ReportingCore } from '../../../server';
 import { cryptoFactory } from '../../../server/lib';
-import {
-  ConditionalHeaders,
-  CreateJobFactory,
-  ESQueueCreateJobFn,
-  RequestFacade,
-} from '../../../server/types';
+import { CreateJobFactory, ESQueueCreateJobFn } from '../../../server/types';
 import { JobParamsDiscoverCsv } from '../types';
+import { ReportingInternalSetup } from '../../../server/core';
 
 export const createJobFactory: CreateJobFactory<ESQueueCreateJobFn<
   JobParamsDiscoverCsv
->> = function createJobFactoryFn(reporting: ReportingCore) {
+>> = function createJobFactoryFn(reporting: ReportingCore, deps: ReportingInternalSetup) {
```

Comment:
I know this repeats a lot in this PR, but I feel like `deps` is not a great descriptor. There are different kinds of dependencies we could get at setup time and start time. In ReportingCore, this is a private member called `pluginSetupDeps`, which is good (the object fields aren't all plugins, but it's ok). Where this object gets passed around, it's probably good to call it `internalSetup`, or `setupDeps`. The idea is, these deps are only available at setup time, which differentiates them from other deps we might use.

Reply:
++, it's a weird "container" of all the various stuff we need, and I thinking calling it what we call the type is good.